UnitsNet/CustomCode/QuantityParser.cs

```diff
@@ -183,6 +183,13 @@ private static string GetRegexPatternForUnitAbbreviations(IEnumerable<string> ab
         private static bool ExtractValueAndUnit(Regex regex, string str, out string valueString, out string unitString)
         {
             var match = regex.Match(str);
+
+            if(match.Groups[0].Value == match.Groups[1].Value || match.Groups.Count < 3)
```

Comment:
A short comment could be useful here, on its own it's hard to read what this actually checks since I don't know the regex looks like at this point and what the groups represent.